elements/pl-order-blocks/pl-order-blocks.py

```diff
@@ -0,0 +1,412 @@
+        colour = 'badge-danger'
+        score = 0
+        feedback = None
+
+        if answer_name in data['submitted_answers']:
+            student_submission = data['submitted_answers'][answer_name]['student_raw_submission']
+        if answer_name in data['partial_scores']:
+            colour = render_html_colour(data['partial_scores'][answer_name]['score'])
+            score = data['partial_scores'][answer_name]['score'] * 100
+            feedback = data['partial_scores'][answer_name]['feedback']
+
+        html_params = {
+            'submission': True,
+            'uuid': uuid,
+            'parse-error': data['format_errors'].get(answer_name, None),
+            'student_submission': prettyPrint(student_submission),
+            'colour': colour,
+            'score': score,
+            'perfect_score': True if score == 100 else None,
+            'feedback': feedback
+        }
+
+        # Finally, render the HTML
+        with open('pl-order-blocks.mustache', 'r', encoding='utf-8') as f:
+            html = chevron.render(f, html_params)
+        return html
+
+    elif data['panel'] == 'answer':
+        if pl.get_boolean_attrib(element, 'external-grader', False):  # if True
+            try:
+                base_path = data['options']['question_path']
+                file_lead_path = os.path.join(base_path, 'tests/ans.py')
+                with open(file_lead_path, 'r') as file:
+                    solution_file = file.read()
+                return f'<pl-code language="python">{solution_file}</pl-code>'
+            except FileNotFoundError:
+                return 'The reference solution is not provided for this question.'
+
+        permutation_mode = pl.get_string_attrib(element, 'permutation-mode', 'html-order')
+        permutation_mode = 'in any order' if permutation_mode == 'any' else 'in the specified order'
+
+        check_indentation = pl.get_boolean_attrib(element, 'check-indentation', False)
+        check_indentation = ', with correct indentation' if check_indentation is True else ''
+
+        if answer_name in data['correct_answers']:
+            html_params = {
+                'true_answer': True,
+                'question_solution': prettyPrint(data['correct_answers'][answer_name]['correct_answers']),
+                'permutation_mode': permutation_mode,
+                'check_indentation': check_indentation
+            }
+            with open('pl-order-blocks.mustache', 'r', encoding='utf-8') as f:
+                html = chevron.render(f, html_params)
+            return html
+        else:
+            return ''
+
+
+def prettyPrint(array):
+    if array is None:
+        return None
+    prettyPrintAnswer = []
+    for text in array:
+        if len(re.findall(r'\$.+\$', text)) == 1:  # used to match text surrounded by $, aka latex text
+            temp = {'text': text, 'render_as_code': False}
+        else:
+            temp = {'text': text, 'render_as_code': True}
```

Comment:
This feels very hacky. What does `render_as_code` do? If the user wants text rendered as code, it feels like it would be better for them to use a `<pl-code>` element inside the `<pl-answer>`?

Reply:
during the summer we didn't figure out the convention to display the answers, so everything was rendered in `code` blocks, but this breaks latex, so the hack was required to render latex properly.

This now irrelevant hack survived 2020, but it could be safely removed. I think it might have to do with why some blocks are not properly rendered in the answer panel

Reply:
Excellent! I love it when simplifying the code also fixes bugs!